src/engine/OptionalJoin.cpp

```diff
@@ -327,6 +327,8 @@ void OptionalJoin::computeSizeEstimateAndMultiplicities() {
   } else {
     _sizeEstimate = multResult * numDistinctResult;
   }
+  // Don't estimate 0 since then upstream thinks we are sure it's zero
```

Comment:
is "upstream" a fixed term in the QLever engine, or what is its meaning here?

Reply:
Not really, with it I mean in the parents of this node in the `QueryExecutionTree` particularly in `Join` when it checks for `isKnownEmpty()`.